### PR TITLE
[NEW] User audit log with sending password changes log by email for admins

### LINKED ISSUE #2995: Notify admin on user password change
Hi!
It would be great that admin receives an e-mail or notification from rocket.cat that someone change his password.

P.S.: I can see it in Logs, but it would be great to receive it automatically.

TY!

Your Rocket.Chat version: (0.26.0)

### PR DESCRIPTION
**Pull request by lindoelio**
@RocketChat/core 

Closes #2995

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
I personally would like to see the password change be added to an audit log, and then the audit log could be emailed at a frequency desired.  

Just feels like this feature is alone is going to be something that so few people use that it won't be properly maintained and one of those areas where bugs will pop up and it'll go un-noticed.

I know if I was running a server I wouldn't want an email telling me about password changes alone.  But one I could scan over with various audit log would be extremely useful.

For sure wouldn't expect this PR to implement all aspects of the audit log just the password change.  

What do you think?

**Comment by reetp** [comment]
@geekgonecrazy that is emminently sensible.

Separate logging/audditing would be much better with a choice of emailing or not as an extra.

**Comment by jszaszvari** [comment]
@geekgonecrazy agreed. Why would you need to be notified upon a password change. A audit log would be better.

**Comment by lindoelio** [comment]
@geekgonecrazy I agree with you.

In this PR, the password change events have been adding to a stored audit log on users collection; and then it can be emailed for admins at a frequency desired, optionally. If the option for this has been disabled, the password change events will continue to be stored for future scan audit normally, persistently stored the most near as possible of account's context.

What do you think about this design?

File: packages/rocketchat-mentions/Mentions.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ export default class {
 				return match;
 			}
 			const name = this.useRealName && mentionObj && mentionObj.name;
+			console.log({name});
```
Comment (gdelavald) [inline packages/rocketchat-mentions/Mentions.js]: This seems like a bad conflict fix. This was removed in a previous commit.

File: packages/rocketchat-mentions-flextab/client/actionButton.js [hunk 2]
```diff
@@ -7,9 +7,6 @@ Meteor.startup(function() {
 		action() {
 			const message = this._arguments[1];
 			RocketChat.MessageAction.hideDropDown();
-			if (window.matchMedia('(max-width: 500px)').matches) {
```
Comment (gdelavald) [inline packages/rocketchat-mentions-flextab/client/actionButton.js]: This seems like a bad conflict fix. This was added in a previous commit.

File: packages/rocketchat-ui/client/views/app/room.js [hunk 3]
```diff
@@ -299,14 +299,6 @@ Template.room.helpers({
 			};
 		});
 		return { buttons };
-	},
```
Comment (gdelavald) [inline packages/rocketchat-ui/client/views/app/room.js]: This seems like a bad conflict fix. This feature was added in a previous commit.

**Comment by geekgonecrazy** [comment]
@lindoelio I think thats a bit better way of handling it. Actions added to audit log... and can choose to have audit logs emailed to you.  I just don't want to add extra complexity like this to the code base for a feature that will not be used by but a few people.

Also, might want to rebase.  It seems like your code is committing back changes we have removed from the code base.

**Comment by lindoelio** [comment]
@geekgonecrazy Actions added to audit log, which can to emailed to admins, as suggested.

@rodrigok and @marceloschmidt Take have a look and review so, please.

To infinity... and beyond! :-)

**Comment by engelgabriel** [comment]
We decided to do a future PR with the Audit Logs part only. We believe that sending emails with LOGs should be handled at the sysadmin level.

**Comment by vynmera** [comment]
I was just thinking about that, audit logs (for all sorts of moderator and useracc actions) would be nice. Perhaps we could even select specific event types to send emails to sysadmins about

**Pull request closed without merging**